### DISCUSSION TITLE
Image + Ping

### DISCUSSION THREAD

**Discussion by Skigamer13** [Rapptz/discord.py#6810] [Q&A] [answered]
Is there a way to make it so that if a user inputs the command with a user mentioned after to have the bot send the image and mention the user? Here is the current code for the command, and any help will be much appreciated. 

if msg.startswith("D$bonk"):
        await message.channel.send(file=discord.File('DogeBot.jpg'))

**Comment by mesub7** [comment, marked as answer]
Hey there,

It's probably best to use the commands extension as it makes working with arguments much easier.

If you did use it, you could pass a `user` argument, which you could typehint to [`discord.Member`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member) (or [`discord.User`](https://discordpy.readthedocs.io/en/latest/api.html#discord.User) if you wanted to use it outside of guilds). 

This would then allow the library to convert the input to a member/user object (assuming it's valid) so that you can mention them with [`member.mention`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.mention) along with sending the file.

You can read about commands here: https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html and typehinting to discord models (such as discord.Member) here: https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#discord-converters

Take care,

**Reply by Skigamer13** [reply]
by the commands extension do you mean "from discord.ext import commands"?

**Reply by mesub7** [reply]
Yes I do. 

You'll want to use `commands.Bot` from the commands extension to handle commands (as shown in the documentation. It's not recommended to use `on_message` to handle commands unless you have a special usecase which the extension hinders for one reason or the other.

**Reply by Skigamer13** [reply]
this is just a part of the code, I have both "import discord" and "from discord.ext import commands" at the top of my code already.

**Reply by Skigamer13** [reply]
also, I should mention that this is my first ever discord bot and most of the code's structure is from following along a tutorial with some revisions by me to give it commands specific to what I want my bot to do.

**Reply by Skigamer13** [reply]
I was also looking through the Discord.py library, and I am having a hard time figuring out which code makes it so a command word sends a predetermined message, as opposed to what the user types after the command, as I want to switch my bot over to the code format in the Discord.py library to make things easier to troubleshoot and add on.

**Reply by mesub7** [reply]
Assuming you want the message to be sent in the same channel that the command is ran in, you would want to use `ctx.send` (a coroutine). 

I would strongly advise reading the commands documentation linked above. To supplement that, please find an official example of a basic bot which utilise the commands extension here' https://github.com/Rapptz/discord.py/blob/master/examples/basic_bot.py

Generally, we discourage tutorials as they don't usually go into how the code actually works and they are often outdated.
